lib/node_modules/@stdlib/string/base/stickycase/test/test.js

```diff
@@ -0,0 +1,79 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var stickycase = require( '@stdlib/string/base/stickycase/lib' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof stickycase, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function throws an error if not provided a string', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		5,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		[],
+		{},
+		function noop() {}
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws a type error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			stickycase( value );
+		};
+	}
+});
+
+tape('the function converts a string to sticky caps case', function test(t) {
+	var actual;
+
+	// Test with default probability (p=0.5)
+	actual = stickycase('hello world');
+	t.equal(typeof actual, 'string', 'returns a string');
+
+	// Test with lower probability (p=0.2)
+	actual = stickycase('hello world', 0.2);
+	t.equal(typeof actual, 'string', 'returns a string');
+
+	// Test with higher probability (p=0.8)
+	actual = stickycase('hello world', 0.8);
+	t.equal(typeof actual, 'string', 'returns a string');
+
+	t.end();
+});
```

Comment:
Once refactored to throw an error when provided a second argument that is not a probability, should add a test here to verify that behavior.